featuretools/tests/entityset_tests/test_es.py

```diff
@@ -1747,6 +1747,98 @@ def test_entityset_equality(es):
 
     first_es.add_relationship('customers', 'id', 'sessions', 'customer_id')
     assert first_es != second_es
+    assert second_es != first_es
 
     second_es.add_relationship('customers', 'id', 'sessions', 'customer_id')
     assert first_es == second_es
+
+
+def test_entityset_id_equality(es):
+    first_es = EntitySet(id='first')
+    fisrt_es_copy = EntitySet(id='first')
+    second_es = EntitySet(id='second')
+
+    assert first_es != second_es
+    assert first_es == fisrt_es_copy
+
+
+def test_entityset_time_type_equality(es):
+    first_es = EntitySet()
+    second_es = EntitySet()
+    assert first_es == second_es
+
+    first_es.time_type = 'numeric'
+    assert first_es != second_es
+
+    second_es.time_type = ltypes.Datetime
+    assert first_es != second_es
+
+    second_es.time_type = 'numeric'
+    assert first_es == second_es
+
+
+@pytest.fixture(params=['make_es', 'dask_es_to_copy'])
+def es_to_copy(request):
+    return request.getfixturevalue(request.param)
+
+
+@pytest.fixture
+def dask_es_to_copy(make_es):
+    es = EntitySet(id=make_es.id)
+    for df in make_es.dataframes:
+        dd_df = dd.from_pandas(df.reset_index(drop=True), npartitions=4)
+        dd_df.ww.init(schema=df.ww.schema)
+        es.add_dataframe(dd_df)
+
+    for rel in make_es.relationships:
+        es.add_relationship(rel.parent_dataframe.ww.name, rel.parent_column.name,
+                            rel.child_dataframe.ww.name, rel.child_column.name)
+    return es
+
+
+def test_deepcopy_entityset(es_to_copy):
+    # Uses make_es since the es fixture uses deepcopy
+    copied_es = copy.deepcopy(es_to_copy)
+
+    assert copied_es == es_to_copy
+    assert copied_es is not es_to_copy
+
+    for df_name in es_to_copy.dataframe_dict.keys():
+        original_df = es_to_copy[df_name]
+        new_df = copied_es[df_name]
+
+        assert new_df.ww.schema == original_df.ww.schema
+        assert new_df.ww._schema is not original_df.ww._schema
+
+        pd.testing.assert_frame_equal(to_pandas(new_df), to_pandas(original_df))
+        assert new_df is not original_df
+
+
+def test_deepcopy_entityset_woodwork_changes(es):
+    if ks and isinstance(es.dataframes[0], ks.DataFrame):
+        pytest.xfail('Cannot deepcopy Koalas DataFrames')
```

Comment:
Unless I'm misunderstanding something, the problem with being unable to deepcopy Koalas DataFrames comes from Koalas. The tests for `EntitySet.concat` xfail for koalas because `"Koalas deepcopy fails"`, so it seems like it's not the end of the world to wait for a potential deepcopy implementation for Koalas. 

But to build a workaround for that we have a few options, and if the change is simple enough, I'm happy to include it in this PR so that we can use concat for koalas EntitySets:

- bring the DataFrame into memory with `to_pandas` and bring back to koalas, which will create a copy of the data, and note that this can be very computationally intensive
- do a shallow copy of just the dataframe (which does work) for koalas and leave a note that the object is the only shallowly copied, so any changes should be made with that in mind. All the other attributes of the entityset would still be deepcopied

Reply:
Since we are not currently supporting concat with Koalas entitysets, so I think it would be best to leave this as a separate issue/PR that we could tackle after the initial integration. Maybe capture your thoughts in a new issue though so we can revisit later?

Reply:
Created an issue: #1466